org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/info/InfoWindow.java

```diff
@@ -2056,6 +2123,30 @@ protected String buildDataSQL(int start, int end) {
 		return dataSql;
 	}
```

Comment:
hi @PeterTakacs300 , since dynaJoint is in InfoPanel and m_sqlMain is created in InfoPanel, it seems it is better to move most of the changes here to InfoPanel instead (the prepareTable method).

Reply:
Hi @hengsin, I think that would not work well. In your case we would make the dynamic JOIN clauses a part of m_sqlMain (so they would not be dynamic anymore). For better performance, we should add them only if the InfoPanel.sort() method is called. But when sorting, InfoPanel.prepareTable() is not called anymore, thats why the parsing is done in InfoWindow.buildDataSQL(). Or am I missing something?

Reply:
ok, that make sense. btw, in what use case would joinTables have > 1 entry ?

Reply:
Oh, you got me with this one, this needs to be fixed. 

That's a remain of an early concept of this solution. We should join only one table at a time since we can not sort multiple columns simultaneously... So currently if I'm sorting more ID-columns one after another, the array collects the tables and creates unnecessary joins.

Fixed in the latest commit.